client/src/App.jsx

```diff
@@ -43,12 +43,13 @@ const App = () => {
     const code = urlParams.get('code');
     const accessTokenResponse = await fetch(`/api/auth/accessToken/${code}`);
     const accessToken = await accessTokenResponse.text();
-
+    
     window.sessionStorage.setItem('accessToken', accessToken);
-
+    
     const userDataResponse = await fetch(`/api/auth/userData/${accessToken}`);
     const userDataJson = await userDataResponse.json();
-
+    
+    window.location.replace('/'); // Is this needed? If so, is there a better way?
     setUserData(userDataJson);
```

Comment:
Hello @DavidAkaFunky!
If you look at what's happening in the frontEnd, the page seems to the user that loads 2 times, since the screen flashes after returning from Login.
This happens because authFromCode is an async function, meaning that setUserData() can happen before reloading the page. A potential fix for this is here:
```js
    await fetch(`/api/auth/userData/${accessToken}`)
      .then((userDataResponse) => {
        window.location.replace('/').then(() => {
          const userDataJson = userDataResponse.json();
          if (userDataJson) setUserData(userDataJson)
        });
      })
```